src/status_im/utils/keychain.cljs

```diff
@@ -0,0 +1,49 @@
+(ns status-im.utils.keychain
+  (:require [cljs.core.async :as async]
+            [taoensso.timbre :as log]
+            [status-im.react-native.js-dependencies :as rn]))
+
+(def username "status-im.encryptionkey")
+
+(defn random-key []
+  (let [length 64
+        arr (make-array length)]
+    (dotimes [i length]
+      (aset arr i (- (rand-int 256) 128)))
+    arr))
+
+(defn- encryption-key-fetch [channel reject]
+  (-> (.getGenericPassword rn/keychain)
+      (.then
```

Comment:
Could we use `.then(fn1).catch(fn2)` instead of `.then(fn1, fn2)` here? 
If there's any error in `fn1` then it will not be running `fn2` using the latter. Same in the code below. What do you reckon?

Reply:
it's the same as JS .then right? I think arg 1 is resolve and arg 2 is reject, .catch is just syntax sugar

Reply:
I believe it's the same as js, but there is a difference between the two:
```
Promise.resolve()
  .then(
     function() { throw "error"}, 
     function() {console.log("never runs")}
)
```
The above will give you an uncaught error, as the error handler is not in the promise chain.

```
Promise.resolve()
.then(function() { throw "error"})
.catch(function() { console.log("caught")})
```
This will catch the error when it happens in the `success` handler.

So unless you actually want errors in the `success` not handled, using `.catch` is a safer option.